extendeddatacrossword_test.go

```diff
@@ -110,3 +111,44 @@ func TestRepairExtendedDataSquare(t *testing.T) {
 		}
 	}
 }
+
+func BenchmarkRepair(b *testing.B) {
+	for _codecType := range codecs {
+		bufferSize := 64
+		ones := bytes.Repeat([]byte{1}, bufferSize)
+		twos := bytes.Repeat([]byte{2}, bufferSize)
+		threes := bytes.Repeat([]byte{3}, bufferSize)
+		fours := bytes.Repeat([]byte{4}, bufferSize)
```

Comment:
why don't we use random data? Is 64 the bufferSize that we want to optimize for? I thought shares will 256 bytes rather. Also the square size should be close to our estimated relevant square sizes. Maybe 3 or 4 different square sizes would make the most sense (almost empty, small, medium, really large).

Reply:
Fixed in 8da717f and 01f2eb3. The numbers aren't the best!

```
BenchmarkRepair/Repairing_16x16_ODS_using_RSGF8-4           	     291	   4112360 ns/op
BenchmarkRepair/Repairing_16x16_ODS_using_LeopardFF8-4      	     259	   4406930 ns/op
BenchmarkRepair/Repairing_16x16_ODS_using_LeopardFF16-4     	     266	   4402912 ns/op
BenchmarkRepair/Repairing_32x32_ODS_using_RSGF8-4           	      63	  18601199 ns/op
BenchmarkRepair/Repairing_32x32_ODS_using_LeopardFF8-4      	      63	  18681030 ns/op
BenchmarkRepair/Repairing_32x32_ODS_using_LeopardFF16-4     	      63	  18583115 ns/op
BenchmarkRepair/Repairing_64x64_ODS_using_RSGF8-4           	      14	  78674819 ns/op
BenchmarkRepair/Repairing_64x64_ODS_using_LeopardFF8-4      	      16	  71389948 ns/op
BenchmarkRepair/Repairing_64x64_ODS_using_LeopardFF16-4     	      15	  71138939 ns/op
BenchmarkRepair/Repairing_128x128_ODS_using_RSGF8-4         	       3	 379398435 ns/op
BenchmarkRepair/Repairing_128x128_ODS_using_LeopardFF8-4    	       4	 282697386 ns/op
BenchmarkRepair/Repairing_128x128_ODS_using_LeopardFF16-4   	       4	 293993599 ns/op
```

Reply:
Side note: this seems to error when I run with a 256 x 256 square (w/ 256-byte shares). Is that expected? I'm assuming some of the codecs have limits. cc @evan-forbes @liamsi

Reply:
Yeah, the FF8 versions have relatively low limits. You can look them up in the particular libraries.

> The numbers aren't the best!

Happy to do some profiling to understand where the most resources are spent. Remember that we need to add back the last good square which won't help with the numbers.

Reply:
I think I can do a bit of profiling as I explore optimizing in #25. It's on the way 🤷. I expect there's room for significant improvement here.

Reply:
>  I expect there's room for significant improvement here.

Yeah, I think there is lots of low-hanging fruits here. Looking forward to the optimizations. Wish I've added the benchmarks before #33 already :-D

Reply:
As a side note, most oif the time seems to be spent on the pre repair sanity checks:
![image](https://user-images.githubusercontent.com/7441766/114238834-88d08880-9985-11eb-9b31-7046e8398525.png)